Add a troubleshooting doc (#177)
openshift/cluster-cloud-controller-manager-operator: docs/user/troubleshooting.md

# Troubleshooting guide

## Common troubleshooting techniques

All Cluster Cloud Controller Manager Operator (CCCMO) components are deployed in `openshift-cloud-controller-manager` namespace. You can check their status by executing `oc get all -n openshift-cloud-controller-manager`.

> To see the operator logs use `oc logs -n openshift-cloud-controller-manager cluster-cloud-controller-manager-operator-<random suffix> -c cluster-cloud-controller-manager`.

Reviewer: i think it would be good to call out this distinction in the opening paragraph

## Cluster Cloud Controller Manager Operator is degraded

> First, check the cluster operator status with `oc get clusteroperator cloud-controller-manager -oyaml` to find potential problems. Most likely the operator will have `Degraded` condition set to True.

Reviewer 2: Nit, add a gap between the flag and argument so it's clear `yaml` is an argument
```suggestion
First, check the cluster operator status with `oc get clusteroperator cloud-controller-manager -o yaml` to find potential problems. Most likely the operator will have `Degraded` condition set to True.
```

## Migration from KCM to CCM got stuck

To troubleshoot the issue we need to understand how the migration works. When `ExternalCloudProvider` feature gate is set, Kube Controller Manager Operator updates all Kube Controller Manager (KCM) pods by setting `--cloud-provider external` there, and starts observing the pods statuses. If all of them started successfully, the operator sets condition `CloudControllerOwner: False` to the `kubecontrollermanager` resource. You can check it with `oc get kubecontrollermanager cluster -o yaml`.

In parallel with that CCCMO also monitors the `kubecontrollermanager` resource. It doesn't provision any Cloud Controller Manager related resources until the condition is set to False there.

When all KCM pods are restarted and KCM operator informs that it doesn't own cloud controllers anymore by setting the condition, CCCMO starts working. It deploys all necessary resources, monitors their statuses, and, if everything is fine, sets the same condition to True in its cluster operator resource. To checks this, execute `oc get clusteroperator cloud-controller-manager -oyaml` to see the value of `CloudControllerOwner` condition.

> When CCCMO sets the condition, the migration is done. Normally it make take up to 15 minutes.

Reviewer 2: Where did the 15 minute value come from?
```suggestion
When CCCMO sets the condition, the migration is done. We expect this to take around 15 minutes.
```
